cp2k_output_tools/blocks/energies.py

```diff
@@ -4,6 +4,23 @@
 
 from .common import FLOAT, BlockMatch
 
+MAIN_ENERGY_RE = re.compile(
+    rf"""
+^\s*Overlap\ energy\ of\ the\ core\ charge\ distribution:\s*(?P<overlap_core>{FLOAT})\n
+\s*Self\ energy\ of\ the\ core\ charge\ distribution:\s*(?P<self_core>{FLOAT})\n
+\s*Core\ Hamiltonian\ energy:\s*(?P<core_hamiltonian>{FLOAT})\n
+\s*Hartree\ energy:\s*(?P<hartree>{FLOAT})\n
+\s*Exchange-correlation\ energy:\s*(?P<xc>{FLOAT})\n
+(\s*Electronic\ entropic\ energy:\s*(?P<electronic_entropic>{FLOAT})\n)?
+(\s*Fermi\ energy:\s*(?P<fermi>{FLOAT})\n)?
+(\s*Dispersion\ energy:\s*(?P<dispersion>{FLOAT})\n)?
```

Comment:
```suggestion
(?:\s*Exchange-correlation\ energy:\s*(?P<xc>{FLOAT})\n)?
(?:\s*Electronic\ entropic\ energy:\s*(?P<electronic_entropic>{FLOAT})\n)?
(?:\s*Fermi\ energy:\s*(?P<fermi>{FLOAT})\n)?
(?:\s*Dispersion\ energy:\s*(?P<dispersion>{FLOAT})\n)?
```
Nowadays I would explicitly ignore the capture groups you are not interested int.
And you may want to make the `Exchange-correlation` optional. Yes, unrealistic in a real calculation, but should it be 0 for some reason (e.g. because someone made a mistake in the code *cough, cough*) the line will not appear at all  and you will not match anything.